### ISSUE TITLE
dependences 

### ISSUE THREAD

**Issue by kailaw138** [Kawue/provim#9]
Trying to have local installation with "conda env create -f environment.yml" , have the following error, which says qt 5.12.5 is incompatible:

![1](https://github.com/Kawue/provim/assets/119678591/74855820-f2dc-4324-ac89-5b1dc696b7d5)

ok then, changing to qt 5.12.9, leads more errors:

![2](https://github.com/Kawue/provim/assets/119678591/bda26d9a-950c-461c-b764-1a04363faa93)

Please tell me the right version of the dependences.

**Comment by Kawue** [comment]
I will try to create this env on my system. However, it may take some days until I have time left.

**Comment by Kawue** [comment]
I installed a clean version of the newest anaconda package and used the described environment command (Powershell, not CMD, but this should not be relevant). My Qt related packages are:

PyQt5==5.15.10
PyQt5-Qt5==5.15.2
PyQt5-sip==12.13.0
PyQtWebEngine==5.15.6
PyQtWebEngine-Qt5==5.15.2

I have no problems with installing or running the environment.

**Comment by kailaw138** [comment]
Thank you for your quick response, I still could not install.

![Capture](https://github.com/Kawue/provim/assets/119678591/2c33179f-26ed-493b-9806-d89c0a43f1b4)

I modified the environment.yml file with the specified version of PyQt5-sip and PyQtWebEngine, but the problem with pyqt  (not PyQt5) persisted.

the Github version:

![orginal](https://github.com/Kawue/provim/assets/119678591/be501004-6a07-4a16-800c-c6784b858ff3)

my modified version:

![modified](https://github.com/Kawue/provim/assets/119678591/89c7a870-9187-41e0-aa79-fa307f52fd47)

PyQt5, PyQt5-Qt5, and PyQtWebEngine-Qt5 are not in the environment.yml, are they required?

Thanks

**Comment by Kawue** [comment]
I think these packages are installed through the others with internal dependencies. You removed `- pip:` in your version. This part tells the installer to use the pip package manager as source instead of any of the other channels. However, pyqt is not available there and requires pip in this case. So please try to keep the environment.yml from this repository and only add the version numbers. This should look like my example below:

```
...
- zstd=1.4.4
- pip:
    - pyqt5-sip=12.13.0
    - pyqtwebengine=5.15.6
```

**Comment by kailaw138** [comment]
Thank you again for your support. Unfortunately, having pip (in the environment.yml) does not work for me:

![Capture](https://github.com/Kawue/provim/assets/119678591/6b5718cb-eb26-4e72-9c3a-fc70f329247b)

in this way ... it simply creates an environment without the pyqt5-sip and pyqtwebengine installed (previously, I simplified the problem)

ok then, force installation with pip (in the created environment):

![Capture2](https://github.com/Kawue/provim/assets/119678591/e62a8e4c-fcc6-4799-85c4-5448d94b3cf8)

Unfortunately, this does not work (also does not work with conda either)

check pip is installed:

![Capture3](https://github.com/Kawue/provim/assets/119678591/2728a302-f8c8-4651-ab2b-afc76c2e6831)

[This also shows you my purpose, to use imzML-to-HDF5 to generate h5 files, pre-process with pyBASIS or ProViM, and segmentation with SmartGate.]

That is why I removed pip from the environment.yml, and so I could narrow down the problem to just pyqtwebengine:

![Capture4](https://github.com/Kawue/provim/assets/119678591/33585dd4-cd73-4739-abe9-a1474d399238)

However, as you said, pyqtwebengine 5.15.6 not available in conda (indeed 5.15.5 also not available)

![Capture5](https://github.com/Kawue/provim/assets/119678591/61e7489c-016e-4ec2-bc97-a915986ab078)

Which version is right? Would 5.15.7 work?

**Comment by Kawue** [comment]
You can try if it works. If not, just remove it.

Alternatively you could try to delete the entries that need pip completely from the environment.yml, then set the environment up without these packages, activate it and with activated environment you could try to install the pip packages manually with `pip install example-package==2.0.1`

Not the clean way but might work.
